src/main/java/se/sundsvall/supportmanagement/service/ErrandAttachmentService.java

```diff
@@ -18,67 +20,65 @@
 		final var attachmentEntity = ofNullable(toAttachmentEntity(errandEntity, errandAttachment)).orElseThrow(() -> Problem.valueOf(BAD_GATEWAY, ATTACHMENT_ENTITY_NOT_CREATED));
-		return attachmentRepository.save(attachmentEntity).getId();
+		final var attachmentId = attachmentRepository.save(attachmentEntity).getId();
+
+		revisionService.createRevision(errandEntity);
+		return attachmentId;
 	}
 
 	public ErrandAttachment readErrandAttachment(String namespace, String municipalityId, String errandId, String attachmentId) {
-		verifyExistingErrand(errandId, namespace, municipalityId);
-
-		final var attachmentEntity =  attachmentRepository.findById(attachmentId).orElseThrow(() -> Problem.valueOf(NOT_FOUND, String.format(ATTACHMENT_ENTITY_NOT_FOUND, attachmentId)));
-		verifyAttachmentBelongsToErrand(errandId, attachmentEntity);
-
-		return toErrandAttachment(attachmentEntity);
+		final var errandEntity = getErrand(errandId, namespace, municipalityId);
+		return ofNullable(errandEntity.getAttachments()).orElse(Collections.emptyList()).stream()
+			.filter(attachment -> attachment.getId().equalsIgnoreCase(attachmentId))
+			.findAny()
+			.map(ErrandAttachmentMapper::toErrandAttachment)
+			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, String.format(ATTACHMENT_ENTITY_NOT_FOUND, attachmentId, errandId)));
 	}
 
 	public List<ErrandAttachmentHeader> readErrandAttachmentHeaders(String namespace, String municipalityId, String errandId) {
-
-		verifyExistingErrand(errandId, namespace, municipalityId);
-
-		return toErrandAttachmentHeaders(attachmentRepository.findByErrandEntityId(errandId));
+		final var errandEntity = getErrand(errandId, namespace, municipalityId);
+		return toErrandAttachmentHeaders(errandEntity.getAttachments());
 	}
 
 	public void deleteErrandAttachment(String namespace, String municipalityId, String errandId, String attachmentId) {
-
-		verifyExistingErrand(errandId, namespace, municipalityId);
-
-		attachmentRepository.deleteById(attachmentId);
-	}
-
-	private void verifyAttachmentBelongsToErrand(final String errandId, AttachmentEntity attachmentEntity) {
-		final var attachmentErrandId = ofNullable(attachmentEntity.getErrandEntity()).map(ErrandEntity::getId).orElse(null);
-
-		if(!Objects.equals(attachmentErrandId, errandId)) {
-			throw Problem.valueOf(NOT_FOUND, String.format(ATTACHMENT_ENTITY_DO_NOT_BELONG_TO_ERRAND, attachmentEntity.getId(), errandId));
-		}
+		final var errandEntity = getErrand(errandId, namespace, municipalityId);
+		final var attachmentEntity = ofNullable(errandEntity.getAttachments()).orElse(Collections.emptyList()).stream()
+			.filter(attachment -> attachment.getId().equalsIgnoreCase(attachmentId))
+			.findAny()
+			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, String.format(ATTACHMENT_ENTITY_NOT_FOUND, attachmentId, errandId)));
+
+		errandEntity.getAttachments().remove(attachmentEntity);
+		attachmentRepository.deleteById(attachmentEntity.getId());
```

Comment:
Do you really need this? Isn't the line above enough? (If it isn't, something is wrong...)